dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/TrackerEventCriteria.java

```diff
@@ -59,17 +59,20 @@ class TrackerEventCriteria extends PagingAndSortingCriteriaAdapter
 
     private String programStage;
 
-    private ProgramStatus programStatus;
+    @ValueOfEnum( enumClass = ProgramStatus.class )
+    private String programStatus;
```

Comment:
Going from an enum to a String in our type is a step back in type safety. I assume this is necessary to capture the value the user actually supplied if its not one of the possible enum values, right?

Reply:
This might not be an ideal solution, i.e. moving from a strongly-typed to weaker typing. Could there be another solution where we generically handle this problem for enums?